### DISCUSSION TITLE
Pull requests with branch protection and status checks implemented by GitHub actions calling other actions does not work

### DISCUSSION THREAD

**Discussion by humbe76** [community/community#56317] [Pull Requests]
### Select Topic Area

Bug

### Body

- We use pull requests to safeguard master to always be ok.
- We use branch protection to run build/test workflow to verify changes are ok.
- To avoid workflow duplication, the reusable parts of the workflow is in a common repo.

We have mostly been using Azure DevOps pipelines for this, and while it is a pain to link into what is failing from the pull request, it works, in that you can specify the pipeline name and the pipeline is run and verified.

When we change to actions though, this stopped working. There are many dubious things here.
  - Firstly, when we try to search for status checks, the list has bad data in it. It looks like it contains data from old pipelines and workflows that no longer exist in any branch of the repository.
  - You don't see anything about the source. Is it an Azure pipeline, a GitHub action? What workflow was this actually taken from, and in what branches does it exist? It is thus hard to verify that you actually picked the correct one.
  - Opposed to using Azure DevOps pipeline, you don't seem to be able to pick a workflow, either by workflow filename or workflow name given within the yaml file. Instead you seem to have to specify a single job within the workflow. But the PR trigger is on the workflow level, and will run all the jobs in the workflow, and I want all the jobs in the workflow to complete. Why can't I specify the workflow rather than a job? What if multiple workflows use the same job names?
  - And in the case of a workflow calling a workflow in another repository, I try to add the name of the job in the calling workflow, but that doesn't work. That locks up the pull request, where it says the job with the name created by joining job names of caller and called workflow has been successfully run, but it doesn't understand it was the run of the job I specified with just the callers name of the workflow , so it has a second status check that it lists has not been run, because it doesn't understand it's the same job.
  
How is this supposed to work?

It would be great if we could see source information when picking a status check. I want to see that I pick that specific GitHub workflow on the master branch, and not an Azure DevOps pipeline one of the devs happened to try and create last month on his feature branch and is long gone, or possibly an old version of the same GitHub workflow from a feature branch that hadn't called another workflow to do the brunt of the work.

Also, I want to be able to specify the workflow rather than a single job, so I can create branch protection rules that doesn't depend on what I call the jobs in my workflow, and so I can run workflows with multiple jobs in them.. Especially since, I'll be running them anyhow since the pull request trigger is on the workflow level. If I want all 3 jobs in one workflow to be status checks for, I don't want to add all 3 jobs to branch protection, as that will likely mean I'll be running all 3 jobs 3 times.

I currently have 2 workarounds..

1. I can name the combined path of the workflow job name of caller and called workflow in branch protection, so pull requests identify the run workflow as the correct one. Then I hardcode names used in the reusable template out in the project which seems sad, and it sounds like it won't work if I have a workflow with multiple jobs in them.

2. I can remove the status check from branch protection, but leave the pull request trigger in the workflow. Sounds like a bug, but then devs creating pull requests have a few seconds window to complete their pull requests before GitHub notice that a workflow should be run. If they do it fast, they can shortcut the check. When the pull_request build is actually started, it is detected, and then they have to wait for it..

**Comment by ricardo-almeida-mck** [comment]
Exactly my use case. 

It feels like I won't be able to set up real governance as the caller can name the job whatever they want.

I would really like to see better handling of reusable workflows in branch protection. Being able to use the exact content that comes after `- uses:` would be ideal (e.g. would allow for tags/branch refs)
